src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,46 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final String DEFAULT_COLOR = "WHITE";
+    private static final int MAX_RANGE = 20;
+    private static final int DEFAULT_RADIUS = 10;
+    private Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int randomNameFigure = new Random().nextInt(FiguresName.values().length);
+        switch (FiguresName.values()[randomNameFigure]) {
+            case CIRCLE:
```

Comment:
Instead of using `FigureName.values()`, you can just generate a random number from 0 to figures count. 

```suggestion
        switch (randomFigureNumber) {
            case 0:
             ....
            case 1:
             ....
```